ironfish-cli/src/commands/wallet/multisig/sign.ts

```diff
@@ -0,0 +1,379 @@
+    }),
+    account: Flags.string({
+      char: 'a',
+      description: 'Name of the account to use for signing the transaction',
+    }),
+    ledger: Flags.boolean({
+      default: false,
+      description: 'Perform operation with a ledger device',
+    }),
+  }
+
+  async collectStrings(
+    item: string,
+    count: number,
+    additionalStrings: string[],
+  ): Promise<string[]> {
+    const array = []
+
+    for (let i = 0; i < count; i++) {
+      const input = await ui.longPrompt(`${item} #${i + 1}`, { required: true })
+      array.push(input)
+    }
+
+    const result = [...array, ...additionalStrings]
+
+    const withoutDuplicates = [...new Set(result)]
+
+    if (withoutDuplicates.length !== result.length) {
+      throw new Error(`Duplicate ${item} found in the list`)
+    }
+
+    return result
+  }
+
+  private async retryStep<T>(stepFunction: () => Promise<T>): Promise<T> {
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      try {
+        const result = await stepFunction()
+        return result
+      } catch (error) {
+        this.log(`An Error Occurred: ${(error as Error).message}`)
+      }
+    }
+  }
+
+  async start(): Promise<void> {
+    const { flags } = await this.parse(SignMultisigTransactionCommand)
+    const client = await this.connectRpc()
+    await ui.checkWalletUnlocked(client)
+
+    let ledger: Ledger | undefined = undefined
+
+    if (flags.ledger) {
+      ledger = new Ledger(this.logger)
+      try {
+        await ledger.connect(true)
+      } catch (e) {
+        if (e instanceof Error) {
+          this.error(e.message)
+        } else {
+          throw e
+        }
+      }
+    }
+
+    let participantName: string
+    if (!flags.account) {
+      participantName = await ui.multisigSecretPrompt(client)
+    } else {
+      participantName = flags.account
+      const account = (await client.wallet.getAccounts()).content.accounts.find(
+        (a) => a === participantName,
+      )
+      if (!account) {
+        this.error(`Account ${participantName} not found`)
+      }
+    }
+
+    const unsignedTransactionInput =
+      flags.unsignedTransaction ??
+      (await ui.longPrompt('Enter the unsigned transaction', { required: true }))
+    const unsignedTransaction = new UnsignedTransaction(
+      Buffer.from(unsignedTransactionInput, 'hex'),
+    )
+    await renderUnsignedTransactionDetails(
+      client,
+      unsignedTransaction,
+      participantName,
+      this.logger,
+    )
+
+    const { commitment, identities } = await this.retryStep(async () => {
+      return this.performRound1(
+        client,
+        participantName,
+        unsignedTransaction,
+        unsignedTransactionInput,
+        ledger,
+      )
+    })
+
+    this.log('\n============================================')
+    this.log('\nCommitment:')
+    this.log(commitment)
+    this.log('\n============================================')
```

Comment:
should we also include an instruction to share the commitment?

Reply:
done